### ISSUE TITLE
Main UI video stream delay

### ISSUE THREAD

**Issue by rabner** [jgow/org.openhab.binding.mqtt.frigatesvr#10]
Frigate ver: 0.14.1 
OpenHab ver: 4.2.2 on RPi 4

I'm getting about 15s delay displaying stream in Main UI as compared to rtsp directly from Frigate. I'm doing it via item state + ".m3u8" in oh-video-card. Birdseye re-stream in Main UI is only delayed couple seconds (via oh-image-card). Is that normal?

![image](https://github.com/user-attachments/assets/ad3632f2-3811-4467-93fb-f706b6a7b145)

**Comment by jgow** [comment]
Yes there can be a delay (in viewed streams; this does not affect detection which is handled by Frigate). The binding itself doesn't do any video processing - all the video stream generation is done by ffmpeg - the binding just serves up the HLS stream components as generated by ffmpeg. It is possible to change the ffmpeg settings via the binding configuration, it may be worth experimenting as it is entirely possible ffmpeg is buffering the stream.

**Comment by rabner** [comment]
Yes, that would make sense. Just thinking out loud, would it be possible to point to frigate directly in openhab items state? For example as an option when 're-stream' is not enabled in camera thing? 

Yesterday I've created modal page with birdseye view and two video streams to see how it behaves. So now when I checked birdseye is working, but both video streams are not. I can't see any ffmpeg process starting in openhab for those cameras.

**Comment by jgow** [comment]
I did include a form of proxy such that devices that can 'see' the OH instance can access the Frigate API via a URL on the OH instance - this was mainly to help facilitate UI implementations - but I did not design this to proxy the video. 

I can add a String channel on each camera Thing that exposes the URL to the video stream on the frigate server - this could be used in a UI running on the OH instance but if this URL was presented to external clients, these clients would also need to be able to 'see' the Frigate server (which is not the case in my setup). This was why I decided against exposing this URL via a channel - in case the Frigate server was on a subnet distinct from the clients of the OH instance.

In this case one could proxy the video streams; the best solution for this would be simply to implement an apache2 server (or whatever-your-favourite-http-server-is) on the OH box and configure it to proxy the video so that OH clients can stream from the OH instance rather than directly from the Frigate server - but then the URLs used to access the video would be on the OH side.
